test/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```diff
@@ -235,5 +235,61 @@ public void ItPublishesAppWhenRestoringToSpecificPackageDirectory()
                 .Should().Pass()
                      .And.HaveStdOutContaining("Hello World");
         }
+
+        [Fact]
+        public void ItFailsToPublishWithNoBuildIfNotPreviouslyBuilt()
+        {
+            var rootPath = TestAssets.CreateTestDirectory().FullName;
+
+            string newArgs = $"console -o \"{rootPath}\"";
+            new NewCommandShim()
+                .WithWorkingDirectory(rootPath)
+                .Execute(newArgs)
+                .Should().Pass();
```

Comment:
This implicitly restores, otherwise the error further down would be that there was no assets file.

Reply:
Correct, `dotnet new` will implicitly restore unless you pass `--no-restore` to it.